crates/ironrdp-server/src/server.rs

```diff
@@ -100,31 +100,41 @@ impl RdpServer {
                         RdpServerSecurity::Tls(acceptor) => acceptor.accept(stream).await?,
                         RdpServerSecurity::None => unreachable!(),
                     });
-                    let (framed, _) = ironrdp_acceptor::accept_finalize(framed, &mut acceptor).await?;
-                    self.client_loop(framed).await?;
+
+                    match ironrdp_acceptor::accept_finalize(framed, &mut acceptor).await {
+                        Ok((framed, result)) => self.client_loop(framed, result).await?,
+                        Err(err) => error!(?err, "accept finalize error"),
+                    };
                 }
 
                 Ok(BeginResult::Continue(framed)) => {
-                    let (framed, _) = ironrdp_acceptor::accept_finalize(framed, &mut acceptor).await?;
-                    self.client_loop(framed).await?;
+                    match ironrdp_acceptor::accept_finalize(framed, &mut acceptor).await {
+                        Ok((framed, result)) => self.client_loop(framed, result).await?,
+                        Err(err) => error!(?err, "accept finalize error"),
+                    };
                 }
 
-                Err(e) => {
-                    error!("connection error: {:?}", e);
+                Err(err) => {
+                    error!(?err, "accept begin error");
```

Comment:
Nit: FYI, the field is recorded as `error` at other places (although we don’t rely on this fact for anything as of today)